scripts/baseline_comparison/baselines.py

```diff
@@ -0,0 +1,167 @@
+            for method, test_error in automl_errors.to_dict().items():
+                rows_automl.append(
+                    ResultRow(
+                        dataset=dataset,
+                        fold=fold,
+                        method=method,
+                        test_error=test_error,
+                        rank=rank_scorer.rank(dataset_fold_name, test_error),
+                        normalized_score=normalized_scorer.rank(dataset_fold_name, test_error),
+                    )
+                )
+    return rows_automl
+
+
+def zeroshot_results(
+        repo: EvaluationRepository, dataset_names: List[str], n_folds: int, rank_scorer, normalized_scorer,
+        ensemble_sizes: List[int] =[1, 20], portfolio_sizes: List[int]=[5, 10, 20, 40, 80]
+) -> List[ResultRow]:
+    """
+    :param ensemble_sizes: number of caruana sizes to consider
+    :param portfolio_sizes: number of portfolio to consider
+    :return: evaluation obtained on all combinations from `ensemble_sizes` and `portfolio_sizes`
+    """
+    dd = repo._zeroshot_context.df_results_by_dataset_vs_automl
+    df_rank = dd.pivot_table(index="framework", columns="dataset", values="score_val").rank()
+    df_rank.fillna(value=np.nanmax(df_rank.values), inplace=True)
+    assert not any(df_rank.isna().values.reshape(-1))
+    rows_zeroshot = []
+    for dataset in tqdm(dataset_names):
+        for portfolio_size in portfolio_sizes:
+            for ensemble_size in ensemble_sizes:
+                taskid = repo.dataset_to_taskid(dataset)
+                train_datasets = [x for x in df_rank.columns if x.split("_")[0] != str(taskid)]
+                indices = zeroshot_configs(-df_rank[train_datasets].values.T, portfolio_size)
+                portfolio_configs = [df_rank.index[i] for i in indices]
+
+                # run best base model and ensemble
+                suffix = f"-{portfolio_size}-{ensemble_size}"
+                if ensemble_size:
+                    suffix += " (ensemble)"
+                test_errors = repo.evaluate_ensemble(
+                    dataset_names=[dataset],
+                    config_names=portfolio_configs,
+                    ensemble_size=ensemble_size,
+                    rank=False,
+                )
+                assert test_errors.shape[0] == 1  # we send one model, we should get one row back
+                for fold in range(n_folds):
+                    test_error = test_errors[0][fold]
+                    dataset_fold_name = f"{repo.dataset_to_taskid(dataset)}_{fold}"
+                    rows_zeroshot.append(ResultRow(
+                        dataset=dataset,
+                        fold=fold,
+                        method=f"Zeroshot{suffix}",
+                        test_error=test_error,
+                        rank=rank_scorer.rank(dataset_fold_name, test_error),
+                        normalized_score=normalized_scorer.rank(dataset_fold_name, test_error),
+                    ))
```

Comment:
consider adding `portfolio_configs` to this object. That way we can start to create the dictionary map for running on the benchmark directly

Reply:
I thought about it after our conversation, I will push a small commit later to include this one.